### DISCUSSION TITLE
Wall clock time in Isaac Gym benchmarks?

### DISCUSSION THREAD

**Discussion by ArthurAllshire** [Toni-SM/skrl#42] [Q&A]
## NVIDIA Isaac Gym

| Environment | PPO |
| --- | --- |
| Allegro Hand <sup>1</sup> | 3942.69 |
| Ant | 5466.3 +/- 279.61 |
| Anymal | 61.86 +/- 1.81 |
| Anymal Terrain | 19.82 +/- 0.57 |
| Ball Balance | 288.07 +/- 25.54 |
| Cartpole <sup>2</sup> | 494.34 +/- 0.87 |
| Franka Cabinet | 3134.0 +/- 182.99 |
| Humanoid | 6474.34 +/- 696.27 |
| Ingenuity | 7066.82 +/- 488.97 |
| Quadcopter | 1237.75 +/- 127.05 |
| Shadow Hand | 7898.38 +/- 180.75 |

| Environment | AMP |
| --- | --- |
| Humanoid | 295.65 +/- 0.86 |	

<hr>

The following charts show the episode's mean length in timesteps (left) and the mean total reward (right)

Allegro Hand (PPO)
![AllegroHand](https://user-images.githubusercontent.com/22400377/193334269-2d3b49c1-1fe0-41a3-9a28-4ee26739943a.png)

Ant (PPO)
![Ant](https://user-images.githubusercontent.com/22400377/193334345-4bdcc8f6-24f9-4336-9db4-44ddf2813c94.png)

Anymal (PPO)
![Anymal](https://user-images.githubusercontent.com/22400377/193334393-9bb1f846-dde5-4162-bcad-78fd9e955358.png)

Anymal Terrain (PPO)
![AnymalTerrain](https://user-images.githubusercontent.com/22400377/193334398-0aa4be9b-5aed-4fbb-9da6-40b223fc2c91.png)

Ball Balance (PPO)
![BallBalance](https://user-images.githubusercontent.com/22400377/193334498-bb7ef523-b8eb-469a-8a5a-02961d6595fb.png)

Cartpole (PPO)
![Cartpole](https://user-images.githubusercontent.com/22400377/193334501-447ded6d-9985-4e8d-bea0-5981d0f1234f.png)

Franka Cabinet (PPO)
![FrankaCabinet](https://user-images.githubusercontent.com/22400377/193334507-01cabd60-e2ab-498a-9ef3-60fdd962ce2d.png)

Humanoid (PPO)
![Humanoid](https://user-images.githubusercontent.com/22400377/193334514-39513050-2e8a-4fe1-bd91-a23d90316781.png)

Humanoid (AMP: imitate different pre-recorded human animations)
![HumanoidAMP](https://user-images.githubusercontent.com/22400377/193334642-4fa18775-5810-4c16-8777-6e9ce73720ed.png)

Ingenuity (PPO)
![Ingenuity](https://user-images.githubusercontent.com/22400377/193334662-ac48cec5-d6ef-45a0-a014-37e9e3d94155.png)

Quadcopter (PPO)
![Quadcopter](https://user-images.githubusercontent.com/22400377/193334664-6b2d4ff8-ff0a-4f22-ad5f-6db17bb4c1e5.png)

Shadow Hand (PPO)
![ShadowHand](https://user-images.githubusercontent.com/22400377/193334667-1edc0d06-bc63-4273-a4af-3bf557b9ceaf.png)

_Originally posted by @Toni-SM in https://github.com/Toni-SM/skrl/discussions/32#discussioncomment-3774815_
 
Hi, I was looking through the benchmark results here (above) for Isaac Gym and was wondering if you could provide also wall clock time in them or if you have the info about how long it took to train each of them? Since for Isaac Gym training that is the critical variable for me to understand the performance due to the ability to vary the number of environments. Thanks :)

**Comment by Toni-SM** [comment]
Hi @ArthurAllshire 

I still have to process the data... but in the following file you can find the time measurements of all the experiments shown in the charts. The experiments were performed on a Quadro RTX 6000 (If I remember well)

[time.txt](https://github.com/Toni-SM/skrl/files/9914185/time.txt)

The time is in seconds and it is measured for the whole script execution (env load  + component initializations + learning)

For example:

```bash
# ppo_cartpole
for i in {1..10}
do
    start=`date +%s`
    python ppo_cartpole.py headless=True
    end=`date +%s`
    echo ppo_cartpole $((end-start)) >> time.txt
done
```

**Comment by Toni-SM** [comment]
Hi @ArthurAllshire 

And, this is the time measurements for Omniverse Isaac Gym

[time.txt](https://github.com/Toni-SM/skrl/files/9918089/time.txt)

**Comment by ArthurAllshire** [comment]
Thanks, this is quite useful.

**Comment by ArthurAllshire** [comment]
By the way do you have plans to add  recurrent training to this library?

**Reply by Toni-SM** [reply]
Yes. In fact, I'm working on it...
